sources/slack/helpers.py

```diff
@@ -0,0 +1,157 @@
+"""Slack source helpers"""
+from typing import Any, Iterable, List, Optional
+from urllib.parse import urljoin
+
+import pendulum
+from dlt.common.time import ensure_pendulum_datetime
+from dlt.common.typing import Dict, TDataItems
+from dlt.sources.helpers import requests
+
+from .settings import MAX_PAGE_SIZE, SLACK_API_URL
+
+
+class SlackApi:
```

Comment:
this is not the best way to work with classes. 
- please rename it to SlackAPI (just my whim) =)
- it makes sense to create an object of the class only once. when creating a source, not a resource. I would remove all these variables from the init, and I would make all auxiliary functions static.
        self.resource = resource
        self.response_path = response_path
        self.params = params or {}
        self.datetime_fields = datetime_fields or []
        self.page_size = page_size
- it would be more logical to pass these variables to the function get_pages()